File: ysda/1_1/f_extremely_round.py
def check(x):
    count = 0
    for t in str(x):
        if t != '0':
            count += 1
    
    return count == 1


N = 999999
ans = []
res = []
for i in range(1, N+1):
    if check(i):
        ans.append(i)

t = int(input())
for i in range(t):
    n = int(input())
    count = 0
    for j, val in enumerate(ans):
        if val <= n:
            count += 1
    
    res.append(count)

for i in res:
    print(i)

File: ysda/1_1_sets_n_dicts/f_extremely_round.cpp
#include <bits/stdc++.h>

using namespace std;

bool check(int x) {
	int count = 0;
	for (auto const &t : to_string(x)) {
		if (t != '0') {
			count++;
		}
	}
	return count == 1;
}

int main() {
	int N = 999999;
	vector<int> ans;
	vector<int> res;
	for (int i = 1; i <= N; ++i) {
		if (check(i)) {
			ans.push_back(i);
		}
	}

	int t, n;
	cin >> t;
	for (int i = 0; i < t; ++i) {
		cin >> n;
		int count = 0;
		for (int j = 0; j < ans.size(); ++j) {
			if (ans[j] <= n) {
				count++;
			}
		}
		res.push_back(count);
	}

	for (auto const &i : res) {
		cout << i << '\n';
	}
	
	return 0;
}
